src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -767,7 +766,12 @@ void GcsActorManager::DestroyActor(const ActorID &actor_id,
         created_actors_.erase(node_it);
       }
     } else {
-      CancelActorInScheduling(actor, task_id);
+      gcs_table_storage_->ActorTaskSpecTable().Get(
+          actor->GetActorID(),
+          [this, actor](Status status, const boost::optional<rpc::TaskSpec> &result) {
+            assert(result.has_value());
+            CancelActorInScheduling(actor, TaskID::FromBinary(result.get().task_id()));
```

Comment:
I am a little concerned this will cause a random race condition. I think it is a pretty big change because at the end of the `DestroyActor`, some of states might not be properly updated.

If task_id is the only field that's required, why don't we actually just embed it into the actor table data? (only task id)

Reply:
Putting task_id in actor table data is acceptable, but for long term, I would prefer to keep it clean.
Let me embed it for now, and remove it in future once we ensure the states being updated correctly.

Reply:
About `task_id` I found it's generated from actor id, like https://github.com/ray-project/ray/blob/master/src/ray/core_worker/core_worker.cc#L1737
So here is a simple(and stable) way for us to obtain task id, instead of getting it from any table.